### PR TITLE
Task 2 dwh

### PR DESCRIPTION
**Pull request by MariDzin**

### CODE REVIEW

File: DWH/task2/Introduction to Data Warehousing and ETL.homework2_task1.docx
Comment (marimldvnv) [inline DWH/task2/Introduction to Data Warehousing and ETL.homework2_task1.docx:1]: There is no such dimension for star:
image
star diagram is correct, just incorrect description.

snowflake schema is incorrect. it should not be similar to 3nf. it should be retrieved from star schema by normalizing a few dimensions. 
That is how snowflake schema looks like 
image

also please provide diagrams as separate files too

File: DWH/task2/Introduction_to_DWH_and_ETL_Business_Template_TASK_2_2.docx
Comment (marimldvnv) [inline DWH/task2/Introduction_to_DWH_and_ETL_Business_Template_TASK_2_2.docx:1]: Not really clear for me the choice of address dimension in your case.
first of all, there should not be two of them. data is loaded from all the source systems to the same table.
secondly, is it an address of who/what? not clear reference to it from sale (looks like it is the location of a sale, but sounds bit illogical)
if address is about customer, that info should be in customer.

When you update model according recommendations, please think how to retrieve 5 dimensions (you may introduce something like payment method, or employee, or delivery method and etc.)

Price per item is rather about product, not the fact (for fact you have sales amount).
image

also please remove fact table from list of dimensions, it is a bit confusing.

as for the grain, please describe it from business perspective too

**Review by marimldvnv — CHANGES_REQUESTED** [review]
Hello Mariam, please see comments

**Comment by MariDzin** [comment]
> Hello Mariam, please see comments

i was not able to receive notifications from git, but I am all in now

File: DWH/task2/Introduction_to_DWH_and_ETL_Business_Template_TASK_2_2.docx
Comment (marimldvnv) [reply DWH/task2/Introduction_to_DWH_and_ETL_Business_Template_TASK_2_2.docx:1]: let's think of status here. do you expect to update fact table rows in order to change the status constantly?
image
in most cases fact table used only for insertions, it is not really convenient to update each row constantly to get last status. I would say that for such cases exists other structure of fact table (example)
image

so if you don't want to introduce such complex solution better go get rid of such columns in your table, which potentially should be updated

Dealsize in fact table- shouldn't it be replaced with dealsize_id which you introduced?

Please extend time dimension with some week number (will you have time cardinality or date cardinality? I would say date would be more efficient)

**Review by marimldvnv — CHANGES_REQUESTED** [review]
Hello, please see comments

**Review by marimldvnv — APPROVED** [review]

**Pull request merged**